synapse/util/caches/lrucache.py

```diff
@@ -730,3 +730,31 @@ def __del__(self) -> None:
         # This happens e.g. in the sync code where we have an expiring cache of
         # lru caches.
         self.clear()
+
+
+class AsyncLruCache(Generic[KT, VT]):
+    """
+    An asynchronous wrapper around a subset of the LruCache API. On it's own
+    this doesn't change the behaviour but allows subclasses that utilize
+    external cache systems that require await behaviour to be created.
+    """
+
+    def __init__(self, *args, **kwargs):  # type: ignore
+        self.lru_cache: LruCache[KT, VT] = LruCache(*args, **kwargs)
```

Comment:
can we name this to make it clear it is a private attribute? 

```suggestion
        self._lru_cache: LruCache[KT, VT] = LruCache(*args, **kwargs)
```